File: specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/examples-js/VirtualHubIpConfigurationList.js
const { NetworkManagementClient } = require("@azure/arm-network");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to Retrieves the details of all VirtualHubIpConfigurations.
 *
 * @summary Retrieves the details of all VirtualHubIpConfigurations.
 * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/examples/VirtualHubIpConfigurationList.json
 */
async function virtualHubRouteTableV2List() {
  const subscriptionId = process.env["NETWORK_SUBSCRIPTION_ID"] || "subid";
  const resourceGroupName = process.env["NETWORK_RESOURCE_GROUP"] || "rg1";
  const virtualHubName = "hub1";
  const credential = new DefaultAzureCredential();
  const client = new NetworkManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (const item of client.virtualHubIpConfiguration.list(
    resourceGroupName,
    virtualHubName,
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

File: specification/network/resource-manager/Microsoft.Network/Network/stable/2025-07-01/examples-java/VirtualHubIpConfigurationList.java

/**
 * Samples for VirtualHubIpConfiguration List.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-07-01/VirtualHubIpConfigurationList.json
     */
    /**
     * Sample code: VirtualHubRouteTableV2List.
     * 
     * @param manager Entry point to NetworkManager.
     */
    public static void virtualHubRouteTableV2List(com.azure.resourcemanager.network.NetworkManager manager) {
        manager.serviceClient().getVirtualHubIpConfigurations().list("rg1", "hub1", com.azure.core.util.Context.NONE);
    }
}
